### ISSUE TITLE
QSelect dialog on small screens is unintuitive and ambiguous - usability issues

### ISSUE THREAD

**Issue by tegola** [quasarframework/quasar#4609]
**Is your feature request related to a problem? Please describe.**
On small screens, QSelect displays its picker as a small dialog instead of a dropdown, as shown on the screenshot below (from [QSelect test page](https://quasar.dev/vue-components/select#Design)).

I see two issues here.

First, there is no way to confirm the selection; I know that just the act of selecting a value confirms it and I just need to tap on the black overlay to hide the dialog, but most people doesn't or will take a while to discover (you can't beat stupid).
While the dialog closes automatically when selecting a value, using the `multiple` props keeps it open. This isn't a real issue with a dropdown, since the black overlay isn't present behind those.

Furthermore – and this is the second issue – the `clearable` props shows an "X" button on the field, which exacerbate the first issue, as it gives the impression it will close the dialog, while it will clear the selected data.

**Describe the solution you'd like**
At its minimum, I'd like a "Done" button at the very end, just like any other confirmation dialog.

I would also ditch the "X" button and use an explicit "Clear" button on the bottom left, but I'm open to other solutions.

**Describe alternatives you've considered**
While I don't like the idea of scrolling to center the view on the options, the ability to force the dropdown instead of the dialog would be enough.

**Additional context**

![select](https://user-images.githubusercontent.com/1868590/60974885-30d99d80-a32b-11e9-8d7c-e399e34a6d73.png)

**Comment by pdanpdan** [comment]
The main problem is that on mobiles, when the keyboard is open, you barely have space for a few options (and on iOS you cannot know and adjust the height for when the keyboard is not visible). So a close button would lower even more the number of visible options.
Also on iOS you practically cannot use the dropdown.

**Comment by tegola** [comment]
Well, since it's already a dialog, How about making it a _maximized_ dialog with a “Clear” button on the top left and a “Done” one on the top right? Having a dialog the size of a dropdown doesn't make much sense to me anyway.

**Comment by rstoenescu** [comment]
@tegola The keyboard on iOS covers the app screen real estate as opposed to Android keyboard which shrinks it. So on iOS you wouldn't be able to see all the options...

**Comment by pdanpdan** [comment]
From my point of view (Android user :) ) the current way QSelect without use-input behaves on mobiles looks very much like the native one. I agree that for iOS the experience is very different from native (in a good way).

There is no native select with filter, so we tried to keep the same ideea (minus the required movement on top because of iOS).

**Comment by tegola** [comment]
@rstoenescu Is this an issue with Cordova and iOS web view, as discussed [here](https://github.com/apache/cordova-ios/issues/472)?

**Comment by pdanpdan** [comment]
It's a problem with iOS, both in browser and Cordova.

**Comment by satyavh** [comment]
There's also an issue when there's a input filter on QSelect, see this screenshot:

![Uploading Screenshot 2019-07-15 at 11.58.42.png…]()

There's no way to close the popup, without selecting an option.

**Comment by pdanpdan** [comment]
@satyavh The image is not visible, please reupload.

**Comment by louisameline** [comment]
I agree with this, I opened an issue not long ago that shared some of these concerns, closed since then by the team (#4565).  

I came here again because I was going to ask if we could have an option to force QSelect to dropdown, instead of a opening dialog. @tegola mentioned the same idea so that's why I'm commenting here.  

My issue is that I have the QSelect in a top dialog: on desktop it's fine, but on mobile it opens two dialogs: 

![rec](https://user-images.githubusercontent.com/371814/61303066-29ad0680-a7e7-11e9-8b2e-3d781f00cfeb.gif)  

I don't think the team can easily solve this. Probably the component is no longer meant for that use case, I'll ditch the QSelect here again. Thanks

**Comment by pdanpdan** [comment]
Do you have only clients/users that use Android? If you want to target iOS users what you want is not possible (not usable).

**Comment by rstoenescu** [comment]
New prop will be available in 1.0.6 for QSelect: `behavior`.

**Comment by 2x2xplz** [comment]
Looking at the latest source code (https://github.com/quasarframework/quasar/blob/1c80106e5abd6a3bdeb217fe33f9b25c23f6673c/ui/src/components/select/QSelect.js), it appears the `behavior` options will be `dialog` or `menu`. Can you please add a `nothing` or `desktop` which just treats the QSelect like it is on a desktop machine (no dialogs or popups, just in-place select). 

Keep in mind this UI "feature" is not documented on the QSelect docs page at all, so it often comes as a surprise to the developers (who are probably developing on a full-sized screen).

**Comment by rstoenescu** [comment]
@2x2xplz hi,

1. Documented at https://quasar.dev/vue-components/select and also the API card.
2. What do you mean by "just in-place select"? You either let QSelect use a dynamic approach ("default"), or force using a "menu" or force using a "dialog".

**Comment by rstoenescu** [comment]
Correct URL: https://quasar.dev/vue-components/select#Options-list-display-mode (edited above too).

**Comment by 2x2xplz** [comment]
Thank you, I had checked the docs and was still seeing 1.0.5. I have upgraded to 1.1 and it appears that `behavior="menu"` prevents the dialogs from popping up on mobile devices. This is what I was hoping for. 

My concern was that `behavior` would provide a choice between a menu (dialog in the middle of the screen) or filter (dialog on top of screen) but not an option to have no dialog at all. 

Great work as always, Quasar continues to get more impressive.
